### PR TITLE
[EHN] Set `expand_column`'s `sep` default as `"|"`

### LINKED ISSUE #876: [ENH] `expand_column` function should set `sep` default is `|`
# Brief Description

Compare to `pandas.Series.str.get_dummies` and `janitor.expand_column` definition.

```python
    def expand_column(
        df: pd.DataFrame, column_name: Hashable, sep: str, concat: bool = True
    ) -> pd.DataFrame:
```

https://github.com/pandas-dev/pandas/blob/5f648bf1706dd75a9ca0d29f26eadfbb595fe52b/pandas/core/strings/accessor.py#L1971

```python
    def get_dummies(self, sep="|"):
```

`get_dummies`'s `sep` parameter default is `|`.
`expand_column`'s `sep` parameter requires the user to input.
And `expand_column` is a sugary syntax of `get_dummies`.

I always forget to add sep when I use this function.  So the result would get a `TypeError`: missing 'sep'.
Why not set `sep` to `|`, same parameter to `get_dummies`?

# Example API

From

```python
    def expand_column(
        df: pd.DataFrame, column_name: Hashable, sep: str, concat: bool = True
    ) -> pd.DataFrame:
```

to

```python
    def expand_column(
        df: pd.DataFrame,
        column_name: Hashable,
        sep: str = "|",
        concat: bool = True,
    ) -> pd.DataFrame:
```

### PR DESCRIPTION
**Pull request by Zeroto521**
# PR Description

Please describe the changes proposed in the pull request:

- Set `expand_column`'s `sep` default is `"|"`, same to `pandas.Series.str.get_dummies`.

**This PR resolves #876.**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.
2. [x] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [x] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

File: tests/functions/test_expand_column.py [hunk 1]
```diff
@@ -25,3 +25,16 @@ def test_expand_and_concat():
         column_name="col1", sep=", ", concat=True
     )
     assert df.shape[1] == 8
+
+
+@pytest.mark.functions
+def test_sep_default_parameter():
```
Comment (ericmjl) [inline tests/functions/test_expand_column.py:31]:
```suggestion
def test_sep_default_parameter():
    """Test that the default parameter is a pipe character `|`."""
```

File: janitor/functions.py [hunk 2]
```diff
@@ -1294,7 +1294,10 @@ def _fill_empty(df, column_names, value=None):  # noqa: F811
 @pf.register_dataframe_method
 @deprecated_alias(column="column_name")
 def expand_column(
-    df: pd.DataFrame, column_name: Hashable, sep: str, concat: bool = True
+    df: pd.DataFrame,
+    column_name: Hashable,
+    sep: str = "|",
+    concat: bool = True,
```
Comment (ericmjl) [inline janitor/functions.py:1298]: I'd like to suggest that we document explicitly the rationale for setting the separator. It's exactly as you wrote in the changelog, which is sufficient for the docstrings!

**Review by ericmjl — CHANGES_REQUESTED** [review]
@Zeroto521 thank you for the contribution! I like the change, as it's a sensible one, and I think we'd also appreciate just a bit more user-facing and developer-facing documentation for it. I only have two requested changes, if you could make those, I'm happy to approve the PR.

As is custom, I would also want to make sure we have two reviewers on the PR before we hit merge. Please hang tight; the team is distributed globally (AU, IN, US, etc.), it might take a bit of time for us to get to it!

**Review by samukweku — APPROVED** [review]

**Comment by ericmjl** [comment]
@samukweku when the checks are done, please feel free to do the honors! (Not worried about code coverage on this one either, the decrease doesn't seem to be reasonable... must be a bug with codecov I think.)

**Review by ericmjl — APPROVED** [review]

**Pull request merged**